qa/tasks/cephfs/test_volumes.py

```diff
@@ -302,6 +305,27 @@ def test_volume_rm(self):
         else:
             raise RuntimeError("expected the 'fs volume rm' command to fail.")
 
+    def test_volume_rm_arbitrary_pool_removal(self):
+        """
+        That the arbitrary pool added to the volume out of band is removed
+        successfully on volume removal.
+        """
+        new_pool = "new_pool"
+        # add arbitrary data pool
+        self.fs.add_data_pool(new_pool)
+        vol_status = json.loads(self._fs_cmd("status", self.volname, "--format=json-pretty"))
+        self._fs_cmd("volume", "rm", self.volname, "--yes-i-really-mean-it")
+
+        #check if fs is gone
+        volumes = json.loads(self._fs_cmd("volume", "ls", "--format=json-pretty"))
+        if (self.volname in [volume['name'] for volume in volumes]):
+            self.fail("Expected the 'fs volume rm' command to succeed. "
+                      "The volume {0} not removed.".format(self.volname))
```

Comment:
Try to prefer using `self.assertFoo` methods. You can restructure as:

````
volnames = [volume['name'] for volume in volumes]
self.assertNotIn(self.volname, volnames)
````

This gives much better error messages if the check fails.

Reply:
Done